### DISCUSSION TITLE
Metrics for projects packages and pagespeed insights have failed

### DISCUSSION THREAD

**Discussion by qinyihao** [lowlighter/metrics#140] [Q&A] [answered]
Hey, I don't know whether it's my problem, but metrics about github packages have failed, which looks like this: ![](https://raw.githubusercontent.com/qinyihao/akioi/main/picgo/20210214203426.png)

Also, the pagespeed insights also returned a warning message which looks like this: ![](https://raw.githubusercontent.com/qinyihao/akioi/main/picgo/20210214203518.png)

Please help me fix that, thanks. I've checked and updated all the API tokens in the `Secrets` area.

**Comment by lowlighter** [comment, marked as answer]
For the projects one I think that's because mix between user account and organization projects (and vice-versa) are not supported.
I'll look at it this week to fix it 👍

As for pagespeed, 500 errors are usually returned when pagespeed api wasn't able to reach your website.
When testing on [web.dev](https://web.dev/measure/) your website I get the following :
```
Error: Error: Lighthouse returned error: FAILED_DOCUMENT_REQUEST. 
Lighthouse was unable to reliably load the page you requested. 
Make sure you are testing the correct URL and that the server is properly responding to all requests. 
(Details: net::ERR_TIMED_OUT)
```

Since your website is up, maybe check if you have any firewall/fail2ban/bot-blocking rules. 
If everything is fine and web.dev is still not able to reach it, unfortunately there's nothing more I can do 😅 

For reference, since my website is behind a Cloudflare proxy, sometimes it flagged the pagespeed bot as a spamming/malicious one so I needed to tune it a bit to make it work

**Comment by qinyihao** [comment]
All right thanks!

**Comment by qinyihao** [comment]
I'd like to say that I didn't fix anything on my website, only by running web.dev several times, I made the pagespeed insights back again.

**Comment by qinyihao** [comment]
interesting though

**Comment by lowlighter** [comment]
Yeah I assume since it's just a timeout, sometimes pagespeed succeed to reach your website and sometimes not.
Maybe some kind of network issue or something like that but I reckon there's not that much waiting time to load your website so it's really strange.

I don't think it's possible to increase timeout of pagespeed api though, so I don't have any solution to offer you atm for this problem 😕

**Comment by lowlighter** [comment]
Commit [`23c25d2`](https://github.com/lowlighter/metrics/pull/124/commits/23c25d21dd0543386835eb9b404e4e410e364c14) from #124 will lets you use projects from organizations owned repositories 🙂

**Comment by qinyihao** [comment]
> Commit [`23c25d2`](https://github.com/lowlighter/metrics/pull/124/commits/23c25d21dd0543386835eb9b404e4e410e364c14) from #124 will lets you use projects from organizations owned repositories 🙂

OK thank you for that
